### ISSUE TITLE
Common constraint: Description metadata should not be empty

### ISSUE THREAD

**Issue by Popolechien** [openzim/libzim#687]

**Comment by kelson42** [comment]
@Popolechien Agree, we will need to change that… but a few other things have to be considered at the same time IMO:
* short/long description /motto new field
* Max size of each field

**Comment by kelson42** [comment]
Meanwhile situation has evolved. `Description` metadata is mandatory (and with a maximum length) and we have now an optional `LonDescription` (with a maximum length as well). See https://wiki.openzim.org/wiki/Metadata

@mgautierfr @veloman-yunkan My questions to you are:
* Do we already check the presence and the length compliances at libzim creator level?
* If « no », should we and why?
* If « no », i guess this ticket concern `zimcheck`?

**Comment by veloman-yunkan** [comment]
> * Do we already check the presence and the length compliances at libzim creator level?

No (this answer applies to any metadata)

> * If « no », should we and why?

IMHO, if the ZIM spec is not just a recommendation, then libzim creator should do its best to ensure that it produces spec-compliant output. But before enforcing those requirements onto our users, we must first improve the management of the ZIM spec. For example, the spec doesn't seem to have releases and versioning - there is no easy way of finding out in which version of the spec a feature or requirement was introduced, which version(s) of the spec a given version of libzim implements/supports, etc.  

> * If « no », i guess this ticket concern `zimcheck`?

zimcheck has to check all requirements documented in the spec for sure.

**Comment by mgautierfr** [comment]
> If « no », should we and why?

I would say that we shouldn't. I still think that libzim is about the binary format (be able to parse the content) and not about the content itself. I have always said the the current libzim specification mixes both and we should somehow split it.
I have back down a bit on this but I'm still reluctant to make libzim too much kiwix oriented. For example libzim has a specific api to read metadata (when you add to read a entry in a specific namespace before) but it doesn't expect any metadata to be present (you still have to give the metadata name) and fail "gracefully" (exception) if the metadata is not present.

Enforcing a Metadata to not be empty (or not more than an arbitrary length) is not the job of libzim as breaking those rules doesn't break the zim file itself.
A bit same that having absolute or external links is breaking kiwix reader but it doesn't break the libzim format.
It is more the work of zimcheck to check that. Saying that, zimcheck is rather a kiwixcheck (not saying we have to move or rename it)

**Comment by kelson42** [comment]
OK, transfering from libzim to zim-tools
